src/main/kotlin/app/batch/Validator.kt

```diff
@@ -0,0 +1,106 @@
+import app.exceptions.BadDecryptedDataException
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.text.SimpleDateFormat
+import java.util.*
+
+@Component
+class Validator {
+
+    fun skipBadDecryptedRecords(item: SourceRecord, decrypted: String): DecryptedRecord? {
+        val hbaseRowKey = Arrays.copyOfRange(item.hbaseRowId, 4, item.hbaseRowId.size)
+        val hbaseRowId = String(hbaseRowKey)
+        val db = item.db
+        val collection = item.collection
+        try {
+            val jsonObject = parseDecrypted(hbaseRowId, decrypted)
+            if (null != jsonObject) {
+                retrieveId(hbaseRowId, jsonObject)
+                val lastUpdatedTimestamp = retrievelastUpdatedTimestamp(hbaseRowId, jsonObject)
+                lastUpdatedTimestamp?.let { validate(hbaseRowId, lastUpdatedTimestamp) }
+                jsonObject.addProperty("timestamp", item.hbaseTimestamp)
+                return DecryptedRecord(jsonObject, db, collection)
+            }
+        } catch (e: Exception) {
+            val ex = BadDecryptedDataException(hbaseRowId, db, collection, e.message!!)
+            DecryptionProcessor.logger.error(ex.message)
+            throw ex
+        }
+        return null
+    }
+
+    fun parseDecrypted(hbaseRowId: String, decrypted: String): JsonObject? {
+        try {
+            val jsonObject = Gson().fromJson(decrypted, JsonObject::class.java)
+            return jsonObject
+        } catch (e: Exception) {
+            val parsingException = "Exception occurred while parsing decrypted db object"
+            throw Exception(parsingException)
+        }
+        return null
+    }
+
+    fun retrieveId(hbaseRowId: String, jsonObject: JsonObject): JsonObject? {
+        try {
+            val id = jsonObject.getAsJsonObject("_id")
+            if (null == id) {
+                val idNotFound = "id not found in the decrypted db object"
+                throw Exception(idNotFound)
+            }
+            return id
+        } catch (e: Exception) {
+            val idNotFoundException = "Exception : ${e.message}"
+            throw Exception(idNotFoundException)
+        }
+        return null
+    }
+
+    fun retrievelastUpdatedTimestamp(hbaseRowId: String, jsonObject: JsonObject): JsonObject? {
+        try {
+            val lastUpdatedTimestamp = jsonObject.getAsJsonObject("_lastModifiedDateTime")
+            if (null == lastUpdatedTimestamp) {
+                val _lastModifiedDateTimeNotFound = "_lastModifiedDateTime not found in the decrypted db object"
+                throw Exception(_lastModifiedDateTimeNotFound)
+            }
+            return lastUpdatedTimestamp
+        } catch (e: Exception) {
+            val _lastModifiedDateTimeException = "Exception : ${e.message}"
+            throw Exception(_lastModifiedDateTimeException)
```

Comment:
Again - why throw-catch-throw - can you just throw?

Reply:
Initial idea was to have customized message being passed  for both cases with in try , but looks like doesnt add much value so deleted now